ui/cap-react/src/antd/collection/CollectionPermissions.js

```diff
@@ -1,105 +1,95 @@
+import CollectionPermissionsColumn from "./CollectionPermissionsColumn";
 
-const CollectionPermissions = ({ permissions }) => {
-  const [permissionsArray, setPermissionsArray] = useState();
-
+const CollectionPermissions = ({ dont, permissions, editable, handlePermissions, defaultPermissions }) => {
+  const [permissionsArray, setPermissionsArray] = useState([]);
   useEffect(() => {
-    if (permissions) {
+    if (dont) {
+      setPermissionsArray(permissions)
+    }
+    else if (permissions) {
       const { permissionsArray } = permissionsPerUser(permissions.toJS());
       setPermissionsArray(permissionsArray);
     }
-  }, [permissions]);
 
-  const renderPermissionType = (e, type) => {
-    let tags = [];
-    if (e.includes(`deposit-schema-${type}`)) {
-      tags.push(
-        <Tooltip title="Drafts">
-          <Tag color="geekblue" style={{ marginRight: 0 }}>
-            D
-          </Tag>
-        </Tooltip>
-      );
-    }
-    if (e.includes(`record-schema-${type}`)) {
-      tags.push(
-        <Tooltip title="Published">
-          <Tag color="purple" style={{ marginRight: 0 }}>
-            P
-          </Tag>
-        </Tooltip>
-      );
+    if (defaultPermissions) {
+      const permissionsMap = {};
+      let { permissionsArray: perms} = permissionsPerUser(defaultPermissions)
+      for (const item of perms) {
+        permissionsMap[item.email] = item.permissions;
+      }
+
+      const mergedArray = permissions.map((item) => ({
+        ...item,
+        permission: permissionsMap[item.email] || [],
+        permissions: permissionsMap[item.email] || [],
+      }));
+
+      setPermissionsArray(mergedArray)
     }
-    return tags.length ? (
-      <Space>{tags}</Space>
-    ) : (
-      <CloseOutlined style={{ color: "lightgray" }} />
-    );
-  };
+
+  }, [permissions, defaultPermissions]);
 
   return permissionsArray && permissionsArray.length > 0 ? (
     <Table
+      size="small"
+      scroll={{x: true}}
       columns={[
         {
           title: "User",
           dataIndex: "email",
           key: "user",
-        },
-        {
-          title: "Type",
-          dataIndex: "type",
-          key: "type",
-          align: "center",
-          render: e =>
-            e === "egroup" ? (
+          render: (e, item) => <Typography.Text >
+           {item.type === "egroup" ? (
               <Tooltip title="egroup">
                 <TeamOutlined />
               </Tooltip>
             ) : (
               <Tooltip title="user">
                 <UserOutlined />
               </Tooltip>
-            ),
+            )}  {e} 
+          </Typography.Text>
         },
         {
           title: "Read",
           dataIndex: "permissions",
           key: "read",
           align: "center",
-          render: e => renderPermissionType(e, "read"),
+          render: (e, item) => <CollectionPermissionsColumn e={e} item={item} type="read" editable={editable} handlePermissions={handlePermissions} />,
```

Comment:
I would still use a function receiving `e` and `item` and returning `<CollectionPermissionColumn />` to make it a bit cleaner since `editable` and `handlePermissions` are always the same